src/integration-test/tests-inter-team.ts

```diff
@@ -0,0 +1,47 @@
+/* tslint:disable:only-arrow-functions variable-name */
+import { expect } from 'chai';
+import routes, { codes } from './routes';
+import { CharlesClients, TeamType } from './types';
+
+const teamTypes: TeamType[] = [
+  'anonymous',
+  'regular',
+  'admin',
+  'open',
+];
+
+export default (
+  clientsFactory: () => Promise<CharlesClients>,
+  _projectName = 'regular-project',
+) => {
+  let clients: CharlesClients;
+  before(async () => {
+    clients = await clientsFactory();
+  });
+  for (const route of routes) {
+    expect(route.accessMatrix.length).to.eq(teamTypes.length);
+    expect(route.accessMatrix.map(x => x.length).reduce((sum, c) => sum + c, 0))
+      .to.eq(teamTypes.length ** 2);
+    describe(route.description, () => {
+      for (let i = 0; i < teamTypes.length; i++) {
```

Comment:
Could some descriptive names be used instead of `i` and `j`? It's a bit hard to follow what a row is and what a column is.